larpandoracontent/LArObjects/LArDiscreteProbabilityVector.cc

```diff
@@ -0,0 +1,237 @@
+/**
+ *  @file   larpandoracontent/LArObjects/LArDiscreteProbabilityVector.cc
+ *
+ *  @brief  Implementation of the lar discrete probability vector class
+ *
+ *  $Log: $
+ */
+
+#include "larpandoracontent/LArObjects/LArDiscreteProbabilityVector.h"
+
+#include <algorithm>
+#include <cmath>
+#include <numeric>
+
+namespace lar_content
+{
+
+template <typename TX, typename TY>
+DiscreteProbabilityVector::DiscreteProbabilityVector(const InputData<TX, TY> &inputData, const TX xUpperBound, 
+    const bool useWidths) :
+    m_xUpperBound(static_cast<float>(xUpperBound)),
```

Comment:
OK, so it has to be possible for any TX to be cast to a float. This is the case for all explicit input types listed below here, for export of functions to library. This addresses a comment I made in the header file about const reference vs. const value.

Reply:
Yeah, right now the methods really limit the input types.  Down the line that might change but for now, conversion to float is a demand.